.github/workflows/requirements.txt

```diff
@@ -0,0 +1 @@
+platformio==6.1.15
```

Comment:
Why is this necessary? Couldn't we go always with the latest here?

Reply:
This is the latest version available via PIP

Reply:
And the requirements.txt file is required for caching using the GitHub Action

Reply:
Looks different than recommended: https://docs.platformio.org/en/stable/integration/ci/github-actions.html

Reply:
It shows the default variant, and I believe it is outdated as it uses setup-python@v4. 
This new caching uses is integrated in the setup-python@v5